### PR TITLE
Fixing broken PyPI publishing

### PR DESCRIPTION
**Pull request by og113**
The issue was that an .hdf5 file was changed by the checkout process (for some reason?) and this meant the tag was out of date. Fixed by adding data in `tests/TestData/N*/*.hdf5` to git-lfs.

### CODE REVIEW

**Review by benoitlaurent96 — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -19,14 +19,30 @@ jobs:
...
       run: |
         python -m pip install --upgrade pip
         pip install build
+
+    - name: Check git status
+      run: |
+        git status
+        git log -n 2
+        git diff
+
```
Comment (niemilau) [inline .github/workflows/publish.yml:40]: What's this for? I'm imagining the intention may be to abort if there are local changes, but I don't think this even does that?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -19,14 +19,30 @@ jobs:
...
+    - name: Install WallGo Python package
+      run: |
+        pip uninstall WallGo
+        pip install . --upgrade -v
```
Comment (niemilau) [inline .github/workflows/publish.yml:44]: Is the uninstall necessary? I'm under the impression that github runners always start with a fresh virtual environment

File: .github/workflows/publish.yml (hunk 1 above)
Comment (og113) [reply .github/workflows/publish.yml:40]: Yes, it just prints things for debugging. Any local changes change the tag and break a release on PyPI. I have found it handy to fix the current issue, and don't see much reason to drop it, as if there is a failure in future we will have more info, and it doesn't exactly take any time to execute.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (og113) [reply .github/workflows/publish.yml:44]: Yes, it is. Again, just for debugging purposes. I could remove it another day, but for now I think let's proceed, as we approach the submission deadline.

**Pull request merged**